tsl/src/nodes/skip_scan/exec.c

```diff
@@ -435,23 +501,53 @@ tsl_skip_scan_state_create(CustomScan *cscan)
 	}
 	state->stage = SS_BEGIN;
 
-	state->num_skip_keys = list_length(cscan->custom_private);
+	state->num_skip_keys = list_length(cscan->custom_private) - 1;
 	state->skip_keys = palloc(sizeof(SkipKeyData) * state->num_skip_keys);
 
+	state->eq_funcs = NULL;
+	state->comp_funcs = NULL;
+	state->comp_strategies = NULL;
+	if (state->num_skip_keys > 1)
+	{
+		state->eq_funcs = palloc(sizeof(FmgrInfo) * (state->num_skip_keys - 1));
+		state->comp_funcs = palloc(sizeof(FmgrInfo) * (state->num_skip_keys - 1));
+		state->comp_strategies = palloc(sizeof(StrategyNumber) * (state->num_skip_keys - 1));
+	}
+
 	ListCell *lc;
 	int i = 0;
 	foreach (lc, cscan->custom_private)
 	{
-		List *skipkeyinfo = (List *) lfirst(lc);
+		if (i < state->num_skip_keys)
+		{
+			List *skipkeyinfo = (List *) lfirst(lc);
 
-		state->skip_keys[i].distinct_col_attnum = linitial_int(skipkeyinfo);
-		state->skip_keys[i].distinct_by_val = lsecond_int(skipkeyinfo);
-		state->skip_keys[i].distinct_typ_len = lthird_int(skipkeyinfo);
-		state->skip_keys[i].nulls = lfourth_int(skipkeyinfo);
-		state->skip_keys[i].sk_attno = list_nth_int(skipkeyinfo, 4);
+			state->skip_keys[i].distinct_col_attnum = linitial_int(skipkeyinfo);
+			state->skip_keys[i].distinct_by_val = lsecond_int(skipkeyinfo);
+			state->skip_keys[i].distinct_typ_len = lthird_int(skipkeyinfo);
+			state->skip_keys[i].nulls = lfourth_int(skipkeyinfo);
+			Assert(state->num_skip_keys == 1 || state->skip_keys[i].nulls == SKIPKEY_NOT_NULL);
+			state->skip_keys[i].sk_attno = list_nth_int(skipkeyinfo, 4);
```

Comment:
In some places we declare an enum for these list positions for clarity, like `DecompressChunkPrivateIndex`, might be useful here as well.